bin/rights_change.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Compare 2 hathifiles and report which items have changed rights
+# from ic/bib in file1 to pd/bib in file2.
+#
+# Invoke thusly:
+# $ bash rights_change.sh f1 f2
+# Results end up in ic_to_pd_YYYYMMDD.txt.
+# Each record in the output are presumed to have changed from ic
+# to pd between the generation of the 2 files.
+# Script starts at the bottom.
+
+run(){
+    f1=$1
+    f2=$2
+    echo "Started"
+    # First, simplify input hathifiles to the data we want.
+    # All the ic/bib records from f1 into one file...
+    cut_sort $f1 ic > cut_sort_ic.tsv
+    # And all the pd/bib records from f2 into another file.
+    cut_sort $f2 pd > cut_sort_pd.tsv
+    # Then compare the 2 simplified files.
+    isodate=`date +'%Y%m%d'`
+    outfile="`pwd`/ic_to_pd_${isodate}.tsv"
+    diff_records cut_sort_ic.tsv cut_sort_pd.tsv > $outfile
+    echo -e "Wrote $outfile"
+    # Remove intermediate files
+    rm cut_sort_ic.tsv cut_sort_pd.tsv
+    echo "Finished"
+}
+
+# Turn a hathifile into fewer cols and sorted matching lines.
+# Matching means: has rights:$rights and reason:bib
+cut_sort(){
+    file=$1
+    rights=$2
+    # Get these cols from the hathifiles:
+    # 1 (id), 3 (rights), 14 (reason), 16 (govdoc),
+    # grep to only get lines matching $rights,
+    # and sort the output.
+    zcat -f $file |
+	cut -f1,3,14,16 |
+	grep -P "\t${rights}\tbib\t[01]$" |
```

Comment:
Would it make sense to add the `cut -f 1,4` here rather than below in `diff_records`?

Reply:
My unspoken reason for splitting this up is so that there's a chance to look at the intermediate files (the ic file and the pd file). Not a very strong reason but was useful while still prototyping.